src/tests/ftest/util/telemetry_test_base.py

```diff
@@ -84,3 +84,181 @@ def verify_telemetry_list(self, with_pools=False):
             self.fail("\n".join(errors))
 
         self.log.info("Test PASSED")
+
+    def get_min_max_mean_stddev(self, suffix, num_targets):
+        """Get four lists of min, max, mean, stddev.
+
+        Sample get_pool_metrics output.
+        {
+            'engine_io_ops_dkey_enum_latency_min': {
+                'wolf-31': {
+                    '0': {
+                        '0': 6,
+                        '1': 13,
+                        '2': 7,
+                        '3': 6,
+                        '4': 6,
+                        '5': 8,
+                        '6': 10,
+                        '7': 12
+                    }
+                },
+                'wolf-32': {
+                    '1': {
+                        '0': 10,
+                        '1': 9,
+                        '2': 7,
+                        '3': 10,
+                        '4': 18,
+                        '5': 9,
+                        '6': 6,
+                        '7': 11
+                    }
+                }
+            },
+            ...
+
+        Args:
+            suffix (str): Metrics suffix for the metric that has min, max,
+                mean, or stddev at the end.
+            num_targets (int): Number of targets.
+
+        Returns:
+            dict: min, max, mean, stddev from each target in a dict.
+                {
+                    "min": [0, 1, 2, 3, 4, 5, 6, 7...],
+                    "max": [0, 1, 2, 3, 4, 5, 6, 7...],
+                    "mean": [0, 1, 2, 3, 4, 5, 6, 7...],
+                    "stddev": [0, 1, 2, 3, 4, 5, 6, 7...]
+                }
+                Note that all the stats should have the same rank order for the
+                verification to work. Not using the sample input values.
+
+        """
+        output = {}
+
+        for stats_value in ["min", "max", "mean", "stddev"]:
+            specific_metrics = [suffix + stats_value]
+            pool_out = self.telemetry.get_pool_metrics(
+                specific_metrics=specific_metrics)
+            self.log.info("pool_out = %s", pool_out)
+
+            values = [0 for _ in range(num_targets)]
+            offset = 0
+
+            for rank_to_dicts in pool_out[specific_metrics[0]].values():
+                for target_to_val in rank_to_dicts.values():
+                    for target, value in target_to_val.items():
+                        values[int(target) + offset * 8] = value
+
+                    offset += 1
```

Comment:
Does `specific_metrics[0]` correspond to rank 0?
What does `8` correspond to?
It seems `int(target) + offset * 8` would overflow the size of `values`. Maybe `8` is the number of targets per engine and `num_targets` refers to the TOTAL number of targets in the system? If so, maybe `8` should be a param or at least a constant, etc.

Reply:
specific_metrics[0] holds the metric string such as 'engine_io_ops_dkey_enum_latency_min'. It's a single-element list because get_pool_metrics() requires a list as input.
Yes, 8 is the number of target per rank and num_targets is the total number of targets. I created a new param and updated the variable name.